data-manipulation/compression/compress-data-via-zlib-inflate-or-deflate.yml

```diff
@@ -0,0 +1,37 @@
+# generated using capa explorer for IDA Pro
+rule:
+  meta:
+    name: compress data via ZLIB inflate or deflate
+    namespace: data-manipulation/compression
+    authors:
+      - blas.kojusner@mandiant.com
+    scope: function
+    references:
+      - https://github.com/madler/zlib/blob/cacf7f1d4e3d44d871b605da3b647f07d718623f/inflate.c#L622
+      - https://github.com/madler/zlib/blob/cacf7f1d4e3d44d871b605da3b647f07d718623f/deflate.c#L763
+    examples:
+      - 6EE9BB8B897C2D69F797646D5F94DE0F:0x65C0
+      - 6EE9BB8B897C2D69F797646D5F94DE0F:0x7930
+  features:
+    - or:
+      - and:
+        - description: Captures artifacts in ZLIB inflate
+        - bytes: 10 05 01 00 17 05 01 01 13 05 11 00 1B 05 01 10 = Distance of fixed Huffman table
+        - bytes: 60 07 00 00 00 08 50 00 00 08 10 00 14 08 73 00 = Length of fixed Huffman table
+      - and:
+        - description: Captures artifacts in ZLIB deflate
+        - operand[1].number: 0x2A = INIT_STATE
+        - basic block:
+          - and:
+            - description: Write ZLIB Header
+            - instruction:
+              - mnemonic: shl
+              - operand[1].number: 0xC = Shift s->w_bits
+            - instruction:
+              - mnemonic: sub
+              - operand[1].number: 0x7800 = Offset to header
+        - operand[1].number: 0x08 = Z_DEFLATED
+        - operand[1].number: 0x06 = level_flags
+        - operand[1].number: 0x20 = PRESET_DICT
+        - operand[1].number: 0x49 = NAME_STATE
+        - characteristic: loop
```

Comment:
im concerned about FPs here, given that many of these numbers are less than 0x100 and therefore are reasonable offsets within structures (except maybe 0x49 and the other odd/unaligned numbers). makes me wish we had that testbest to be able to show the prevalence of rule/branch matches.

Reply:
I'll run this against a bunch of frozen features.

Reply:
Found 5 matches across 85 samples from our testbed.

```
tests/data/112f9f0e8d349858a80dd8c14190e620.exe_
tests/data/276f691a3df25481f59d79781799e35f.exe_
tests/data/39c05b15e9834ac93f206bc114d0a00c357c888db567ba8f5345da0529cbed41.dll_
tests/data/42e81cc1145ba3c1936a6cf9b8da0ccd.dll_
tests/data/50d5ee1ce2ca5e30c6b1019ee64eeec2.exe_
```

@bkojusner if you have a moment, please review these. I can also take a look if you're busy.

Reply:
All of these were true positives.

Reply:
ok, great! that gives me more confidence that we should merge as-is. thanks!